examples/ex18.cpp

```diff
@@ -250,60 +253,259 @@ int main(int argc, char *argv[])
+      case 3:
+         *mesh_file = "../data/periodic-square.mesh";
+         break;
+      case 4:
+         *mesh_file = "../data/periodic-segment.mesh";
+         break;
+      case 5:
+         *mesh_file = "../data/periodic-square.mesh";
+         break;
+      default:
+         throw invalid_argument("Default mesh is undefined");
+   }
+}
+
+// Initial condition
+VectorFunctionCoefficient EulerInitialCondition(const int problem,
+                                                const double specific_heat_ratio,
+                                                const double gas_constant)
+{
+   switch (problem)
+   {
+      case 1: // fast moving vortex
+         return VectorFunctionCoefficient(4, [specific_heat_ratio,
+                                              gas_constant](const Vector &x, Vector &y)
+         {
+            MFEM_ASSERT(x.Size() == 2, "");
+
+            double radius = 0, Minf = 0, beta = 0;
+            // "Fast euler"
+            radius = 0.2;
+            Minf = 0.5;
+            beta = 1. / 5.;
+
+            const double xc = 0.0, yc = 0.0;
+
+            // Nice units
+            const double vel_inf = 1.;
+            const double den_inf = 1.;
+
+            // Derive remainder of background state from this and Minf
+            const double pres_inf = (den_inf / specific_heat_ratio) *
+                                    (vel_inf / Minf) * (vel_inf / Minf);
+            const double temp_inf = pres_inf / (den_inf * gas_constant);
+
+            double r2rad = 0.0;
+            r2rad += (x(0) - xc) * (x(0) - xc);
+            r2rad += (x(1) - yc) * (x(1) - yc);
+            r2rad /= (radius * radius);
+
+            const double shrinv1 = 1.0 / (specific_heat_ratio - 1.);
+
+            const double velX =
+               vel_inf * (1 - beta * (x(1) - yc) / radius * exp(-0.5 * r2rad));
+            const double velY =
+               vel_inf * beta * (x(0) - xc) / radius * exp(-0.5 * r2rad);
+            const double vel2 = velX * velX + velY * velY;
+
+            const double specific_heat =
+               gas_constant * specific_heat_ratio * shrinv1;
+            const double temp = temp_inf - 0.5 * (vel_inf * beta) *
+                                (vel_inf * beta) / specific_heat *
+                                exp(-r2rad);
+
+            const double den = den_inf * pow(temp / temp_inf, shrinv1);
+            const double pres = den * gas_constant * temp;
+            const double energy = shrinv1 * pres / den + 0.5 * vel2;
+
+            y(0) = den;
+            y(1) = den * velX;
+            y(2) = den * velY;
+            y(3) = den * energy;
+         });
+      case 2: // slow moving vortex
+         return VectorFunctionCoefficient(4, [specific_heat_ratio,
+                                              gas_constant](const Vector &x, Vector &y)
+         {
+            MFEM_ASSERT(x.Size() == 2, "");
+
+            double radius = 0, Minf = 0, beta = 0;
+            // "Slow euler"
```

Comment:
Does this case differ from "Fast Euler" only by `M_inf`? If so, it would be nice to de-duplicate the code.